rails/app/views/search/get_current_material_unassigned_collections.html.haml

```diff
@@ -0,0 +1,91 @@
+.search_materials
+  - if @material.length == 1
+    - material = @material[0]
+    - if material.is_a? ::Investigation
+      - investigation_name = material.name
+      - activity_name = ""
+      - external_activity_name = ""
+      - existing_collections = nil
+    - elsif material.is_a? ::Activity
+      - investigation_name = ""
+      - external_activity_name = ""
+      - parent_name = (material.parent) ? material.parent.name : ""
+      - activity_name = material.name
+      - existing_collections = []
+      - if material.parent
+        - existing_collections = MaterialsCollection.joins(:materials_collection_items).where(:materials_collection_items => {:material_id=>material.parent.id,:material_type=>"Investigation"})
+    - elsif material.is_a? ::ExternalActivity
+      - investigation_name = ""
+      - activity_name = ""
+      - external_activity_name = material.name
+      - existing_collections = MaterialsCollection.joins(:materials_collection_items).where(:materials_collection_items => {:material_id=>material.id,:material_type=>"ExternalActivity"})
+  - else
+    - investigation_name = ""
+    - parent_name = (@material[0].parent) ? @material[0].parent.name : ""
+    - activity_name = @material.collect{|item| item.name}.join('<br/>')
+
+  - activity_name = activity_name || ""
+  - external_activity_name = external_activity_name || ""
+  - if investigation_name.length > 0
```

Comment:
@emcelroy we often do include code blocks like this in our haml, but it might be better to try and create this data somewhere else, possibly in the controller, because it's difficult to debug and test here in the view I think. 

Let's wait to hear from @scytacki  -- Maybe some of the checking about what kind of material this is can be dropped completely, and this view will get simpler.

Because many (all?) of the material types all respond to `name` I am not sure you need to do a lot of this. 

Also if view logic gets complicated like this, you could also try creating a plain old ruby object which responds to all of the methods you need to display in your view, such as `name`.  Then this class or object can be tested for logic, and the view can just know that the object will respond to a single set of messages to get the values you need in your view.

Reply:
Right since e don't need to support the Activity and investigation types this can be simplified. Additionally (based on the code above) I don't think we need to support the case of `@material.length != 1`. So that simplifies it further.